### PR TITLE
Use npm instead of Yarn

### PR DESCRIPTION
**Pull request by nickmccurdy**

### CODE REVIEW

**Comment by patrickhulce** [comment]
thanks @nickmccurdy! why though? this removes any package lock mechanism 😅

**Comment by nickmccurdy** [comment]
This organization uses npm without a package lock for libraries, and with a package lock for apps. I'm just making pull requests to keep things consistent for contributors.

**Comment by patrickhulce** [comment]
Ah sorry I missed that discussion is there a link anywhere to the rationale? :)

**Comment by nickmccurdy** [comment]
In the gitignores of other libraries: https://github.com/testing-library/dom-testing-library/blob/master/.gitignore#L6-L7

I believe Kent has a more in-depth explanation somewhere but I couldn't find it.

**Comment by patrickhulce** [comment]
I see, thanks! 

I can understand the library dependency angle, but I'm not sure I agree with the rationale when working with devDeps that actually produce distributable assets like pptr-testing-library does though (which is much more similar to the app situation than library dependencies). Removing the lockfile here feels a bit like policy matching for the sake of policy matching rather than actually following the reasoning of the policy :/

How would you feel about replacing with a package lock?

**Comment by nickmccurdy** [comment]
Are you referring to how dependent projects would install versions of pptr-testing-library dependencies? If so, that doesn't matter, as `package-lock.json` and `yarn.lock` are not used to install published packages, only local packages. If a dependency version needs to be locked, `package.json` can have specific versions or a `npm-shrinkwrap.json` can be used (don't think Yarn supports it though).

**Comment by patrickhulce** [comment]
I'm aware of how lock files work :) 

I am not referring to dependent projects. pptr-testing-library in CI [builds a bundle of javascript](https://github.com/testing-library/pptr-testing-library/blob/master/rollup.config.js) that is distributed as part of our own npm package. The lock file ensures that there are no unexpected changes to this bundle and given that we do string replacement and relatively hacky things to make this work, I am concerned that unexpected transitive dependency changes could result in breakages.

I understand the spirit of the ban on lock files, that a library author should ensure their library can handle unexpected dependency versions that real users might install. This package goes [out of its way to ensure we handle different versions](https://github.com/testing-library/pptr-testing-library/blob/fe290fca9789d68cedf914974cfc947c925ab342/.travis.yml#L10-L17), above and beyond what simply removing a lock file would do. Enforcing a lock file ban here would increase the volatility and reproducibility of published versions of this package while not actually reaping any of the stated benefits beyond the steps we've already taken.

I appreciate that you're seeking consistency across the organization, but consistency in reasoning for positive outcomes should prevail over consistency for file existence.

**Comment by nickmccurdy** [comment]
Alright, I'll add a lock file for npm in this case.

We have two options, what would you prefer?
1. Use npm 6, which is stable and used by more users, guessing dependency versions from `node_modules` in a way that may conflict
2. Use npm 7, which is in preview and requires an explicit install for now, which can accurately import versions from `yarn.lock`

**Comment by patrickhulce** [comment]
Thanks @nickmccurdy!

> Use npm 7, which is in preview and requires an explicit install for now, which can accurately import versions from yarn.lock

This sounds great :)

File: .npmignore [hunk 1]
```diff
@@ -2,5 +2,3 @@ coverage/
 .nyc_output/
 .vscode/
 test/
-
-yarn.lock
```
Comment (patrickhulce) [inline .npmignore:6]: should we replace this with `package-lock.json` since, as you note, it won't be respected by downstream consumers?

**Review by patrickhulce — APPROVED** [review]
thanks @nickmccurdy! 🎉

File: .npmignore (hunk 1 above)
Comment (nickmccurdy) [reply .npmignore:6]: For whatever reason, I don't typically see lock files getting ignored, so I figured we could just leave them alone.

**Comment by nickmccurdy** [comment]
This should work locally after running `npm install` with `npm install --global npm@7` (or `npx npm@7 install` to use it temporarily without upgrading your global npm). At this point I'm waiting to see if CI will pass as I'm not sure if Travis is bundling npm 7 yet.

File: .npmignore (hunk 1 above)
Comment (patrickhulce) [reply .npmignore:6]: I suspect that's more due to not thinking about it than a cogent argument ;)

but agreed it's not a big deal

**Comment by nickmccurdy** [comment]
It seems like npm 7 is successfully installed now. Can we get another review please?

**Review by patrickhulce — APPROVED** [review]
still LGTM :)

**Pull request merged**